sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsMessagingAutoConfiguration.java

```diff
@@ -121,16 +122,29 @@ public EventHubsProducerFactory defaultEventHubsNamespaceProducerFactory(
         }
 
         /**
-         * Creates an Event Hubs message converter.
+         * Creates an Event Hubs message converter with an isolated ObjectMapper.
          *
          * @return An Event Hubs message converter.
          */
         @Bean
         @ConditionalOnMissingBean
-        public EventHubsMessageConverter eventHubsMessageConverter() {
+        @ConditionalOnProperty(value = "spring.cloud.azure.message-converter.isolated-object-mapper.enabled", havingValue = "true", matchIfMissing = true)
+        EventHubsMessageConverter eventHubsMessageConverter() {
             return new EventHubsMessageConverter();
         }
 
+        /**
+         * Creates an Event Hubs message converter with ObjectMapper from Spring context.
+         * @param objectMapper An ObjectMapper from Spring context.
+         * @return An Event Hubs message converter.
+         */
+        @Bean
+        @ConditionalOnMissingBean
+        @ConditionalOnProperty(value = "spring.cloud.azure.message-converter.isolated-object-mapper.enabled", havingValue = "false")
```

Comment:
I prefer to call it `spring.cloud.azure.message-converter.isolated-object-mapper`